### ISSUE TITLE
Problem with python-ldap and ldaps://

### ISSUE THREAD

**Issue by abretaud** [galaxyproject/starforge#130]
The current python-ldap wheel doesn't work with secured connection to ldap server (ldaps://)
Galaxy issue: https://github.com/galaxyproject/galaxy/issues/3178

The error is:

```
Traceback (most recent call last):
File "/home/galaxy/galaxy-dist/lib/galaxy/auth/providers/ldap_ad.py", line 126, in authenticate
l.simple_bind_s()
File "/home/galaxy/galaxy-dist/.venv/local/lib/python2.7/site-packages/ldap/ldapobject.py", line 223, in simple_bind_s
resp_type, resp_data, resp_msgid, resp_ctrls = self.result3(msgid,all=1,timeout=self.timeout)
File "/home/galaxy/galaxy-dist/.venv/local/lib/python2.7/site-packages/ldap/ldapobject.py", line 514, in result3
resp_ctrl_classes=resp_ctrl_classes
File "/home/galaxy/galaxy-dist/.venv/local/lib/python2.7/site-packages/ldap/ldapobject.py", line 521, in result4
ldap_result = self._ldap_call(self._l.result4,msgid,all,timeout,add_ctrls,add_intermediates,add_extop)
File "/home/galaxy/galaxy-dist/.venv/local/lib/python2.7/site-packages/ldap/ldapobject.py", line 106, in _ldap_call
result = func(*args,**kwargs)
SERVER_DOWN: {'desc': "Can't contact LDAP server"}
```

Debugging this is not easy... If I find some time I'll try to setup a small docker test system and see what's wrong.

**Comment by wholtz** [comment]
Here is some testing code I wrote that failed on the current python-ldap wheel, but was able to successfully connect when used with a locally built version of python-ldap.

```python
#!/usr/bin/env python
import ldap, sys
LDAP_SERVER = 'ldaps://ldap.example.com'
LDAP_BASE = 'ou=people,dc=example,dc=com'
BIND_USER = 'uid=myname,ou=people,dc=example,dc=com'
BIND_PASSWORD = 'mypassword'

ldap.set_option(ldap.OPT_DEBUG_LEVEL, 4095)

try:
    l = ldap.initialize(LDAP_SERVER)
except ldap.LDAPError, e:
    sys.stderr.write("Fatal Error\n")
    raise

try:
  l.protocol_version = ldap.VERSION3
  l.simple_bind_s(BIND_USER, BIND_PASSWORD)
  result = l.search_s(LDAP_BASE, ldap.SCOPE_SUBTREE)
  print result
  print "OK"
  l.unbind()
except Exception, error:
  print error
```

The version of python-ldap used in both cases was v2.4.27, and v2.7.12 of python. This was on a Ubuntu 16.04 system with the libsasl2-dev python-dev libldap2-dev libssl-dev libsasl2-dev packages.

**Comment by natefoo** [comment]
Fixed in #146. Thanks @abretaud!
